Content.Server/_Goobstation/StationEvents/GameDirectorSystem.cs

```diff
@@ -0,0 +1,576 @@
+                var ruleSecondEnt = GameTicker.AddGameRule(randomAntagSecond.ID);
+
+                GameTicker.StartGameRule(ruleFirstEnt);
+                GameTicker.StartGameRule(ruleSecondEnt);
+            }
+    }
+
+    /// <summary>
+    ///   Count the active players and ghosts on the server.
+    ///   Players gates which stories and events are available
+    ///   Ghosts can be used to gate certain events (which require ghosts to occur)
+    /// </summary>
+    public PlayerCount CountActivePlayers() // DeltaV - Allow admins to force the story to change
+    {
+        var allPlayers = _playerManager.Sessions.ToList();
+        var count = new PlayerCount();
+        foreach (var player in allPlayers)
+        {
+            // TODO: A
+            if (player.AttachedEntity != null)
+            {
+                // TODO: Consider a custom component here instead of HumanoidAppearanceComponent to represent
+                //        "significant enough to count as a whole player"
+                if (HasComp<HumanoidAppearanceComponent>(player.AttachedEntity))
+                    count.Players += 1;
+                else if (HasComp<GhostComponent>(player.AttachedEntity))
+                    count.Ghosts += 1;
+            }
+        }
+
+        return count;
+    }
+
+    /// <summary>
+    ///   Count all the players on the server.
+    /// </summary>
+    public int GetTotalPlayerCount(IList<ICommonSession> pool)
+    {
+        var count = 0;
+        foreach (var session in pool)
+        {
+            if (session.Status is SessionStatus.Disconnected or SessionStatus.Zombie)
+                continue;
+
+            count++;
+        }
+
+        return count;
+    }
+
+    /// <summary>
+    ///   Sorts the possible events and then picks semi-randomly.
+    ///   when maxRandom is 1 it's always the "best" event picked. Higher values allow more events to be randomly selected.
+    /// </summary>
+    protected RankedEvent SelectBest(List<RankedEvent> bestEvents, int maxRandom)
+    {
+        var ranked = bestEvents.OrderBy(ev => ev.Score).Take(maxRandom).ToList();
+
+        var rand = _random.NextFloat();
+        rand *= rand; // Square it, which leads to a front-weighted distribution
+                      // Of 3 items, there is (50% chance of 1, 36% chance of 2 and 14% chance of 3)
+        rand *= ranked.Count - 1;
+
+        var rankedEvent = ranked[(int) Math.Round(rand)];
+
+        // Pick this event
+        var events = String.Join(", ", ranked.Select(r => r.PossibleEvent.StationEvent));
+        LogMessage($"Picked {rankedEvent.PossibleEvent.StationEvent} from best events (in sequence) {events}");
+        return rankedEvent;
+    }
+
+    private void LogMessage(string message, bool showChat=true)
+    {
+        // TODO: LogMessage strings all require localization.
+        _adminLogger.Add(LogType.GameDirector, showChat?LogImpact.Medium:LogImpact.High, $"{message}");
```

Comment:
Why put a string in a format string...
```suggestion
        _adminLogger.Add(LogType.GameDirector, showChat ? LogImpact.Medium : LogImpact.High, message);
```